component/src/main/java/io/siddhi/extension/io/file/FileAlterationImpl.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file;
+
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.extension.io.file.util.Status;
+import org.apache.commons.io.monitor.FileAlterationListener;
+import org.apache.commons.io.monitor.FileAlterationObserver;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.siddhi.extension.io.file.util.Util.getFileHandlerEvent;
+
+/**
+ * Implementation of  FileAlterationImpl
+ */
```

Comment:
This sentence should end with a period. Shall we format that? Please refer to the following about how to write proper doc comments.

https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html

Reply:
https://github.com/siddhi-io/siddhi-io-file/tree/5f2348a3cec4571d14c1f3f939dfddcbadd466e8